File: tasks/seeds/vcfarch-0057/files/design.go
package vcfarch

import "errors"

// Architecture is the machine-readable VCF design returned by Build.
// Implement the concrete JSON model required by the fixtures and schemas.
type Architecture struct{}

// Build constructs the greenfield SddcSpec and the existing-estate migration
// plan from the supplied scenario, inventory, and pinned compatibility snapshot.
func Build(scenarioPath, estatePath, snapshotPath string) (Architecture, error) {
	return Architecture{}, errors.New("VCF architecture builder is not implemented")
}

File: tasks/seeds/vcfarch-0052/files/vcf_architecture/design.py
"""Architecture builder."""

from __future__ import annotations

from typing import Any


def build_architecture(
    inventory: dict[str, Any], compatibility: dict[str, Any]
) -> dict[str, Any]:
    """Build an installer SddcSpec with its architecture extension."""
    raise NotImplementedError("build the VCF 9.0 architecture")
